Build failure: inochi-creator

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- Attempt to build or run Inochi Creator (`inochi-creator`) from Nixpkgs 25.11 / nixos-unstable
- Observe the CMake 4 error seen in other packages with a similar build process.

### Can Hydra reproduce this build failure?

The last build job was manually cancelled.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/9xi60anz87sn142lz8c693wyycfa5f9k-inochi-creator-0.8.6.drv' failed with exit code 2;
       last 25 log lines:
       > -- Detecting C compile features
       > -- Detecting C compile features - done
       > -- Detecting CXX compiler ABI info
       > -- Detecting CXX compiler ABI info - done
       > -- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
       > -- Detecting CXX compile features
       > -- Detecting CXX compile features - done
       > -- submodulePrefixPath:
       > -- Found Freetype: /nix/store/psvw7vs0h8kfggxjxlscsv8hl9189768-freetype-2.13.3/lib/libfreetype.so (found version "2.13.3")
       > -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
       > -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
       > -- Found Threads: TRUE
       > -- Found SDL2: /nix/store/dpz3h60m1p5zvxzfa51vc9bm4bayjyvq-sdl2-compat-2.32.56/lib/libSDL2main.a;/nix/store/dpz3h60m1p5zvxzfa51vc9bm4bayjyvq-sdl2-compat-2.32.56/lib/libSDL2.so
       > CMake Error at cimgui/CMakeLists.txt:1 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       > 
       > -- Configuring incomplete, errors occurred!
       > Error Command failed with exit code 1: cmake -DSTATIC_CIMGUI= -S deps -B deps/build_linux_x64_cimguiStatic
```

### Additional context

Sub-issue of #445447 

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.7, NixOS, 25.11 (Xantusia), 25.11.20250919.8eaee11`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - nixpkgs: `/nix/store/gg9fmyvsizavv6x9srffip7kky023x07-source`

### Notify maintainers

@TomaSajt 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/inochi2d/generic.nix
+++ b/pkgs/applications/misc/inochi2d/generic.nix
@@ -92,6 +92,10 @@ buildDubPackage (
       # We get around this by manually pre-fetching the submodule and copying it into the right place
       cp -r --no-preserve=all ${cimgui-src}/* "$cimgui_dir/deps/cimgui"
 
+      # bump cmake minimum version to a version supported by cmake 4
+      substituteInPlace "$cimgui_dir/deps/cimgui/CMakeLists.txt" \
+          --replace-fail "cmake_minimum_required(VERSION 3.1)" "cmake_minimum_required(VERSION 3.10)"
+
       # Disable the original cmake fetcher script
       substituteInPlace "$cimgui_dir/deps/CMakeLists.txt" \
           --replace-fail "PullSubmodules(" "# PullSubmodules(" \

